src/web/mod.rs

```diff
@@ -167,6 +167,18 @@ impl Handler for MainHandler {
             }
         }
 
+        fn pass_iron_errors_with_redirect(e: IronError) -> IronResult<Response> {
+            // in some cases the iron router will return a redirect as an `IronError`.
+            // Here we convert these into an `Ok(Response)`.
+            if e.error.downcast_ref::<TrailingSlash>().is_some()
+                || e.response.status == Some(status::MovedPermanently)
+            {
+                Ok(e.response)
```

Comment:
Why will there still be a response if there's an error? Also, why does this also handle MovedPermanently? That seems like a change in behavior.

Reply:
I'm not sure I follow. 

In our own code we do redirects by returning an `Ok(Response)`. Our code expects this and passes this through. 

For this specific case (_URL without slash at the end has a match_, see [the iron source](https://github.com/iron/iron/blob/cde77e5e0e9bfa1bf1230f4bdd8491f4f4cdc72c/router/src/router.rs#L171-L195)), iron returns an `IronError(Response)`. 

Since for us the redirect is fine, and not an error, I want to convert this. 

( of course I could be missing something)

Reply:
Ok, I think I see - we have our own error handler on *top* of iron's built-in error handlers. And our error handler expects to recognized every possible error, that's what the last `or_else` on line 197 is doing. Since iron has its own built-in error handlers, some of its errors (but not ours) will actually be valid HTTP responses that we can serve without changes.